tests/ut/ir/operators/test_mx_quant_ops.py

```
@@ -0,0 +1,155 @@
+        k = ir.ConstInt(64, DataType.INDEX, span)
+        src = ir.Var("src", ir.TileType([m, k], DataType.FP32), span)
+
+        with pytest.raises(ValueError, match="requires static M and K"):
+            ir.op.tile.tquant(src, span=span)
+
+    @pytest.mark.parametrize("shape", [[8, 32], [15, 64], [16, 3744]])
+    def test_tquant_rejects_isa_shape_constraints(self, shape):
+        span = ir.Span.unknown()
+        src = ir.Var("src", ir.TileType(shape, DataType.FP32), span)
+
+        with pytest.raises(ValueError, match="M divisible by 16|M\\*K <= 59461"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)